lib/puppet/type/deferred_resources.rb

```diff
@@ -72,7 +79,68 @@
 
     validate do |value|
```

Comment:
shouldn't this be at the top? otherwise the `value` has already been touched in the `munge`